### ISSUE TITLE
Ejercicio listo para su revisión

### ISSUE THREAD

**Issue by JuanMaRuizH** [DevAppWeb/temperature-data-JuanMaRuizH#1]
Un saludo

**Comment by jivanrod** [comment]
Hola Juan Manuel,

 Buen trabajo con el ejercicio que ya va cogiendo un poco de complejidad. Te adjunto cometarios.

- Me da un error un poco extraño a la hora de mostrar la tabla final, en el último foreach las variables $key y $value mantienen los valores asignados en el foreach de la parte inicial del script ?¿, quizás se trate de una bug del entorno de Netbeans.

- Compara la organización del código de tu ejercicio con la solución propuesta en mi repositorio, seguro que ves evidente las ventajas del patrón.

- Es muy útil conocer lo mejor posible las capacidades de la libreria de funciones de PHP, la función sum ya existe en PHP como array_sum

- La única parte de funcionalidad que me falta es la parte de la ordenación. Échale un vistazo a la función array_multisort para ordenar arrays bidimensionales por una combinación de criterios.

Saludos

**Comment by JuanMaRuizH** [comment]
Muchas gracias Jorge, revisaré el error que me comentas, por si el error fuera de mi código.

No he aplicado el patrón MVC porque me acado de dar cuenta con tu post en el foro que también había que aplicarlo en este ejercicio. Intentaré hacer antes de mirar la solución. 

Sobre la organización, se me ha pasado por completo, sorry.

Un saludo.
JuanMa.

**Comment by jivanrod** [comment]
En realidad el enunciado no solicitaba la aplicación del patrón MVC, te lo comento porque después de estructurar la aplicación enlazando formularios y scripts seguro que ves mejor las ventajas de organizar las áreas funcionales de la aplicación en controlador(es), modelo(s) y vista(s)
Un saludo

**Comment by JuanMaRuizH** [comment]
Sí, por su puesto :D

He corregido un par de cosas y he aplicado el patrón.

No he visto nada raro en el foreach de la tabla final a mi sí que me muestra los resultados correctamente.

La aplicación del patrón está hecha en la rama ```temperature-data``` pero he creado una nueva rama con alguna pequeña diferencia en cuanto a organización. He creado un controller para cada formulario, dicho controller está asociado a la vista correspondiente. ```index.php``` es el encargado de decidir qué vista y qué controller carga en cada momento.

La idea que me ha llevado a esta organización ha sido la de separar responsabilidades ya que cada formulario realmente trata datos diferentes (aunque enlazados). Adems quería evitar las duplicidades y he realizado un único inlcude con los partials y la vista necesaria.

Un saludo.
JuanMa.

**Comment by jivanrod** [comment]
Efectivamente, la idea tiene mucho que ver con la separación de funciones en el código. Lo veo bien aunque mi única pega es que no has unificado el código controlador en un único fichero. De esta manera hacemos que todas las peticiones de la aplicación vayan dirigidas a un único punto de entrada (index.php). Dentro del controlador tendremos que saber identificar la petición y como tratarla, terminando con la vista adecuada que le da respuesta.

Saludos
